app/main.py

```diff
@@ -1,3 +1,79 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import os
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+
+    file_name = "config.json"
+    file_name = f"app/{file_name}" \
+        if not os.path.exists(file_name) \
+        else file_name
+
+    with open(file_name, "r") as file:
+        config = json.load(file)
+
+    fuel_price = config["FUEL_PRICE"]
+    customers_data = config["customers"]
+    shops_data = config["shops"]
+
+    customers = []
+    shops = []
+
+    for customer_data in customers_data:
+        car = Car(
+            customer_data["car"]["brand"],
+            customer_data["car"]["fuel_consumption"]
+        )
+
+        customer = Customer(
+            customer_data["name"],
+            customer_data["product_cart"],
+            customer_data["location"],
+            customer_data["money"], car
+        )
+
+        customers.append(customer)
+
+    for shop_data in shops_data:
+
+        shop = Shop(
+            shop_data["name"],
+            shop_data["location"],
+            shop_data["products"]
+        )
+        shops.append(shop)
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        best_shop = None
+        # best_cost = int("inf")
+        best_cost = 100000
```

Comment:
Instead of using a hardcoded value for `best_cost`, consider using `float('inf')` to represent infinity. This is a more standard and clear way to initialize a variable for comparison with costs.